vernac/declare.ml

```diff
@@ -1989,67 +2077,28 @@ let control_only_guard { proof; pinfo } =
     with Exit -> ()
 
-let close_proof ?warn_incomplete ~opaque ~keep_body_ucst_separate ps =
-
-  let { using; proof; initial_euctx; pinfo } = ps in
-  let { Proof_info.info = { Info.udecl } } = pinfo in
-  let { Proof.poly } = Proof.data proof in
-  let elist, uctx = prepare_proof ?warn_incomplete ps in
-  let opaque = match opaque with
-    | Vernacexpr.Opaque -> true
-    | Vernacexpr.Transparent -> false in
-
-  let make_entry ((body, eff), typ) =
-    let keep_body_ucst_separate = if keep_body_ucst_separate then Some initial_euctx else None in
-    let _, univs, body =
-      make_univs_immediate ~poly ?keep_body_ucst_separate ~opaque ~uctx ~udecl ~eff body (Some typ) in
-    definition_entry_core ?using ~univs ~types:typ body
-  in
-  let entries = CList.map make_entry elist in
-  { entries; uctx; pinfo }
-
-type closed_proof_output = (Constr.t * Evd.side_effects) list * UState.t
-
-let close_proof_delayed ~feedback_id ps (fpl : closed_proof_output Future.computation) =
-  let { using; proof; initial_euctx; pinfo } = ps in
-  let { Proof_info.info = { Info.udecl } } = pinfo in
-  let { Proof.poly; entry; sigma } = Proof.data proof in
-
-  (* We don't allow poly = true in this path *)
-  if poly then
-    CErrors.anomaly (Pp.str "Cannot delay universe-polymorphic constants.");
-
-  (* Because of dependent subgoals at the beginning of proofs, we could
-     have existential variables in the initial types of goals, we need to
-     normalise them for the kernel. *)
-  let nf = Evarutil.nf_evars_universes (Evd.set_universe_context sigma initial_euctx) in
-
-  (* We only support opaque proofs, this will be enforced by using
-     different entries soon *)
-  let make_entry i (_, _, types) =
-    (* Already checked the univ_decl for the type universes when starting the proof. *)
-    let univs = UState.univ_entry ~poly:false initial_euctx in
-    let typ = nf (EConstr.Unsafe.to_constr types) in
-
-    (* NB: for Admitted proofs [fpl] is not valid (raises anomaly when forced) *)
-    Future.chain fpl (fun (pf, uctx) ->
-        let (body, eff) = List.nth pf i in
-        (* Deferred proof, we already checked the universe declaration with
-             the initial universes, ensure that the final universes respect
-             the declaration as well. If the declaration is non-extensible,
-             this will prevent the body from adding universes and constraints. *)
-        let uctx = make_univs_deferred_private_mono ~initial_euctx ~uctx ~udecl body (Some typ) in
-        ((body, uctx), eff))
-    |> delayed_definition_entry ?using ~univs ~types:typ ~feedback_id
-  in
-  let entries = CList.map_i make_entry 0 (Proofview.initial_goals entry) in
-  { entries; uctx = initial_euctx; pinfo }
+let return_proof p = (prepare_proof p : closed_proof_output)
 
-let close_future_proof = close_proof_delayed
+let bool_of_opacity = function
+  | Vernacexpr.Opaque -> true
+  | Vernacexpr.Transparent -> false
```

Comment:
It's not clear from the name which input should produce which bool

maybe rename `opacity_is_opaque`? or keep it inlined

Reply:
no
just keep it inlined